File: .claude/hooks/file-changed-reload-env.mjs
#!/usr/bin/env node
// @hookstack file-changed-reload-env
// Recharge les variables d'env d'un fichier modifié dans CLAUDE_ENV_FILE (FileChanged)
import { readFileSync, appendFileSync } from 'fs';
import { fileURLToPath } from 'url';

export function run(
  input,
  {
    readFile = readFileSync,
    append = appendFileSync,
    envFile = process.env.CLAUDE_ENV_FILE,
  } = {},
) {
  if (!envFile || input.event === 'unlink') return null;

  try {
    const content = readFile(input.file_path, 'utf8');
    const lines = content
      .split('\n')
      .filter((l) => /^[A-Z_a-z][A-Z_a-z0-9]*=/.test(l) && !l.startsWith('#'));
    for (const line of lines) {
      append(envFile, `export ${line.trim()}\n`);
    }
    return { count: lines.length, message: `[file-changed-reload-env] reloaded ${lines.length} vars from ${input.file_path}\n` };
  } catch (e) {
    return { error: e.message, message: `[file-changed-reload-env] ${e.message}\n` };
  }
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result?.message) process.stderr.write(result.message);
}

File: .claude/hooks/file-changed-reload-env.py
#!/usr/bin/env python3
# @hookstack file-changed-reload-env
"""Reloads env vars from a changed file into CLAUDE_ENV_FILE (FileChanged)."""
import json
import os
import re
import sys


def _read_file(path):
    with open(path, "r", encoding="utf8") as f:
        return f.read()


def _append(path, content):
    with open(path, "a", encoding="utf8") as f:
        f.write(content)


def run(input_data, *, read_file=None, append=None, env_file=None):
    if read_file is None:
        read_file = _read_file
    if append is None:
        append = _append
    if env_file is None:
        env_file = os.environ.get("CLAUDE_ENV_FILE")

    if not env_file or input_data.get("event") == "unlink":
        return None

    try:
        content = read_file(input_data.get("file_path"))
        lines = [
            l
            for l in content.split("\n")
            if re.match(r"^[A-Za-z_][A-Za-z0-9_]*=", l) and not l.startswith("#")
        ]
        for line in lines:
            append(env_file, f"export {line.strip()}\n")
        return {
            "count": len(lines),
            "message": f"[file-changed-reload-env] reloaded {len(lines)} vars from {input_data.get('file_path')}\n",
        }
    except Exception as e:
        return {
            "error": str(e),
            "message": f"[file-changed-reload-env] {e}\n",
        }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result and result.get("message"):
        sys.stderr.write(result["message"])
